File: 0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {

        var input = s.toCharArray();
        Stack<Character> myStack = new Stack<>();
        for (char c: input) {
            if (c == '(' ) {
                myStack.push(')');
            } else if (c == '{') {
                myStack.push('}');
            } else if(c == '[') {
                myStack.push(']');
            }
            if(c == ')' || c == '}' || c == ']') {
                if (myStack.isEmpty()) {
                    return false;
                }
                var topChar = myStack.pop();
                if (topChar != c) {
                    return false;
                }
            }
        }
        return myStack.isEmpty();
    }
}

File: 0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mapping = {")":"(", "}":"{", "]":"["}

        for char in s:
            if char in "({[":
                stack.append(char)
            elif char in mapping and stack and stack[-1] == mapping[char]:
                stack.pop()
            else:
                return False
        return len(stack) == 0
